cli/src/commands/duplicate.rs

```diff
@@ -72,7 +69,7 @@ pub(crate) struct DuplicateArgs {
         value_name = "REVSETS",
         add = ArgValueCandidates::new(complete::all_revisions)
     )]
-    destination: Vec<RevisionArg>,
+    destination: Option<Vec<RevisionArg>>,
```

Comment:
Why did this have to change? Is there a difference between `None` and `Some(vec![])` now? Oh, looks like it was changed this way for `jj rebase` due to @yuja's comment [here](https://github.com/jj-vcs/jj/pull/4640#discussion_r1800289570). I tried dropping these `Option` wrappings and I found it simpler, fwiw. The closure in `compute_commit_location()` can still return an `Option` for easier and safer pattern matching. I can send a PR once this has been merged to show what it looks like. We can discuss there.

Reply:
My feeling is that `Option<Vec<_>>` encodes the intent better if an empty value is logically different than unspecified (e.g. use default instead of empty list.)